source/backend/api/Areas/DocumentGeneration/DocumentGenerationController.cs

```diff
@@ -54,17 +55,50 @@ public async Task<IActionResult> GetSupportedDocumentTypes()
         }
 
         /// <summary>
-        /// Uploads the passed document.
+        /// Uploads the passed document as a template.
         /// </summary>
         [HttpPost("upload_template")]
-        [HasPermission(Permissions.DocumentAdd)]
+        //[HasPermission(Permissions.GenerateDocuments)]
         [ProducesResponseType(typeof(ExternalResult<string>), 200)]
         [SwaggerOperation(Tags = new[] { "document-generation" })]
         public async Task<IActionResult> UploadTemplate([FromForm] IFormFile file)
         {
             var result = await _documentGenerationService.UploadFileTemplate(file);
             return new JsonResult(result);
         }
+
+        /// <summary>
+        /// Renders the given template with the request data and returns the result as wrapped encoded base64 file.
+        /// </summary>
+        [HttpPost("render_template/download-wrapped")]
```

Comment:
why mix x_y and x-y formatting in the url?

Reply:
yikes